homeassistant/components/plex/sensor.py

```diff
@@ -55,55 +57,30 @@ def __init__(self, plex_server):
         )
         self.hass.data[PLEX_DOMAIN][DISPATCHERS][server_id].append(unsub)
 
-    @callback
-    def async_refresh_sensor(self, sessions):
+    async def async_refresh_sensor(self, sessions):
         """Set instance object and trigger an entity state update."""
-        self.sessions = sessions
-        self.async_schedule_update_ha_state(True)
-
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return self._name
-
-    @property
-    def unique_id(self):
-        """Return the id of this plex client."""
-        return self._unique_id
-
-    @property
-    def should_poll(self):
-        """Return True if entity has to be polled for state."""
-        return False
-
-    @property
-    def state(self):
-        """Return the state of the sensor."""
-        return self._state
-
-    @property
-    def unit_of_measurement(self):
-        """Return the unit this state is expressed in."""
-        return "Watching"
+        _LOGGER.debug("Refreshing sensor [%s]", self.unique_id)
 
-    @property
-    def icon(self):
-        """Return the icon of the sensor."""
-        return "mdi:plex"
+        self.sessions = sessions
 
-    @property
-    def device_state_attributes(self):
-        """Return the state attributes."""
-        return {content[0]: content[1] for content in self._now_playing}
+        @callback
+        def update_plex(_):
+            dispatcher_send(
```

Comment:
We need to change to `async_dispatcher_send` here since we're async context.

Reply:
Addressed in #33785.